pandas/_typing.py

```diff
@@ -84,10 +86,10 @@
 
 # scalars
 
-PythonScalar = Union[str, int, float, bool]
+PythonScalar = Optional[Union[str, int, float, complex, bool, date, time, timedelta]]
```

Comment:
instead of Optional, can you use type(None) to be more explict?

Reply:
should just do a pass to modernize this entire file